From a075f4e09bf836508172a989eac4cb48d956430a Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Tue, 1 Oct 2024 18:20:19 +0200
Subject: [PATCH 1264/1695] dts: stm32: add GPU OPP description in
 stm32mp235.dtsi

Add the description for GPU operating points for stm32mp235 plateform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7f7186808936b81f83b4f93addfe564ae13eceaf dts: stm32: add GPU OPP description in stm32mp235.dtsi]
Change-Id: Ifdc65de2f498d4fa9987925226b4eafd86db5102
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/408153
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp235.dtsi | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp235.dtsi b/core/arch/arm/dts/stm32mp235.dtsi
index 48c862161..87a51b1ef 100644
--- a/core/arch/arm/dts/stm32mp235.dtsi
+++ b/core/arch/arm/dts/stm32mp235.dtsi
@@ -7,4 +7,31 @@
 #include "stm32mp233.dtsi"
 
 / {
+	gpu_opp_table: opp-table-gpu {
+		compatible = "operating-points-v2";
+
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <800000>;
+			opp-supported-hw = <0x3>;
+			st,opp-default;
+		};
+
+		opp-900000000 {
+			opp-hz = /bits/ 64 <900000000>;
+			opp-microvolt = <900000>;
+			opp-supported-hw = <0x2>;
+			st,opp-default;
+		};
+	};
+};
+
+&rifsc {
+	gpu: gpu@48280000 {
+		compatible = "vivante,gc";
+		gpu-supply = <&vddgpu>;
+		clocks = <&rcc CK_BUS_GPU>;
+		operating-points-v2 = <&gpu_opp_table>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_GPU_ID>;
+	};
 };
-- 
2.47.2

